Source/JavaScriptCore/dfg/DFGMovHintRemovalPhase.cpp

```diff
@@ -59,16 +58,64 @@ class MovHintRemovalPhase : public Phase {
     {
         dataLogIf(DFGMovHintRemovalPhaseInternal::verbose, "Graph before MovHint removal:\n", m_graph);
 
+        // First figure out where various locals are live. We only need to care when we exit.
+        // So,
+        // 1. When exiting, mark all live variables alive. And we completely clear dead variables at that time.
+        // 2. When performing MovHint, it is def and it kills the previous live variable.
+        BlockMap<Operands<bool>> liveAtHead(m_graph);
+        BlockMap<Operands<bool>> liveAtTail(m_graph);
+
+        for (BasicBlock* block : m_graph.blocksInNaturalOrder()) {
+            liveAtHead[block] = Operands<bool>(OperandsLike, block->variablesAtHead, false);
+            liveAtTail[block] = Operands<bool>(OperandsLike, block->variablesAtHead, false);
+        }
+
+        bool changed;
+        do {
+            changed = false;
+            for (BlockIndex blockIndex = m_graph.numBlocks(); blockIndex--;) {
```

Comment:
Since we run this to fixed point, it makes sense that the order doesn't matter. I have to wonder though, would it be faster (i.e it would converge in fewer iterations) if we do it in reverse post order? I guess it depends on which phases come before, maybe it is worth it to do only when we don't need to calculate the RPO?